server/internal/interface/grpc/handlers/arkservice.go

```diff
@@ -138,49 +138,29 @@ func (h *handler) GetBoardingAddress(
 func (h *handler) RegisterIntent(
 	ctx context.Context, req *arkv1.RegisterIntentRequest,
 ) (*arkv1.RegisterIntentResponse, error) {
-	notesInputs := req.GetNotes()
 	bip322Signature := req.GetBip322Signature()
 
-	if len(notesInputs) <= 0 && bip322Signature == nil {
+	if bip322Signature == nil {
 		return nil, status.Error(codes.InvalidArgument, "missing inputs")
 	}
 
-	if bip322Signature != nil && len(notesInputs) > 0 {
-		return nil, status.Error(codes.InvalidArgument, "cannot mix vtxos and notes")
+	signature, err := bip322.DecodeSignature(bip322Signature.Signature)
```

Comment:
always use the getters:
```suggestion
	signature, err := bip322.DecodeSignature(bip322Signature.GetSignature())
```

please, apply also to the rest of this function